netlogo-core/src/main/fileformat/LabLoader.scala

```diff
@@ -17,10 +17,10 @@ object LabLoader {
 
 import LabLoader._
 
-class LabLoader(literalParser: LiteralParser, editNames: Boolean = false, existingNames: Set[String] = Set[String]()) {
+class LabLoader(literalParser: LiteralParser) {
   if (literalParser == null)
     throw new Exception("Invalid lab loader!")
-  def apply(xml: String): Seq[LabProtocol] = {
+  def apply(xml: String, editNames: Boolean, existingNames: Set[String]): Seq[LabProtocol] = {
```

Comment:
It looks like the arguments in `class LabLoader `are missing in this version

Reply:
Isaac clarified

> i see the confusion, basically i initially did it as a constructor parameter, but that didn’t work because each labloader object is constructed once and then reused multiple times. since each time it’s used the “existingNames” parameter might be different, i moved the parameters to the apply method instead. i kept the “editNames” parameter for posterity to make sure it worked as it did before the move, but I realize i might be able to remove that parameter so i will check on that now.
> 
> actually the editNames parameter is necessary, since sometimes the existingNames parameter will start empty and fill up as experiments are loaded, so there is no other way to indicate whether names should be changed or not.